include/nbl/builtin/hlsl/bit.hlsl

```diff
@@ -102,6 +108,13 @@ uint16_t countl_zero(T n)
     return impl::clz<sizeof(T)*8>(n);
 }
 
+// TODO: templated "switch" depending on signedness  
+template<typename Unsigned>
+Unsigned bitfield_extract_unsigned( Unsigned val, uint32_t offsetBits, uint32_t numBits )
+{
+    return spirv::bitfieldExtractUnsigned<Unsigned>( val, offsetBits, numBits );
+}
+
```

Comment:
move this to the `glsl` namespace header and conform to
https://registry.khronos.org/OpenGL-Refpages/gl4/html/bitfieldExtract.xhtml

Reply:
btw if you want to "template switch" use a struct  and partial specialization as the implementation behind the scenes

Reply:
how do I use bitfieldExtract in hlsl/cpp "common" code ? glsl:: does not implement any cpp functions. Is it ok to do this or do I just move the hlsl that uses it out of the "common" scope ?

Reply:
create an alias a for the GLM function that does in the `ifndef __HLSL_VERSION` `else` case

Reply:
ok just provide the C++ alias, and we can merge